workflow/controller/operator.go

```diff
@@ -1034,37 +1031,30 @@ func printPodSpecLog(pod *apiv1.Pod, wfName string) {
 // assessNodeStatus compares the current state of a pod with its corresponding node
 // and returns the new node status if something changed
 func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatus) *wfv1.NodeStatus {
```

Comment:
there is a lot of code duplication in this func, specifically it duplicates `makeNodePhase` - it has resulted in bugs